src/main/java/org/openmaptiles/layers/Landuse.java

```diff
@@ -103,10 +109,36 @@ public void process(Tables.OsmLandusePolygon element, FeatureCollector features)
       if ("grave_yard".equals(clazz)) {
         clazz = FieldValues.CLASS_CEMETERY;
       }
-      features.polygon(LAYER_NAME).setBufferPixels(BUFFER_SIZE)
+      var feature = features.polygon(LAYER_NAME).setBufferPixels(BUFFER_SIZE)
         .setAttr(Fields.CLASS, clazz)
-        .setMinPixelSizeOverrides(MIN_PIXEL_SIZE_THRESHOLDS)
         .setMinZoom(Z6_CLASSES.contains(clazz) ? 6 : 9);
+      if (FieldValues.CLASS_RESIDENTIAL.equals(clazz)) {
+        feature
+          .setMinPixelSize(0.1)
+          .setPixelTolerance(0.25);
+      } else {
+        feature
+          .setMinPixelSizeOverrides(MIN_PIXEL_SIZE_THRESHOLDS);
+      }
+    }
+  }
+
+  @Override
+  public List<VectorTile.Feature> postProcess(int zoom,
+    List<VectorTile.Feature> items) throws GeometryException {
+    if (zoom < 6 || zoom > 12) {
+      return items;
+    } else {
+      // merging only merges polygons with class "residential" for z6-z12
+      Map<Boolean, List<VectorTile.Feature>> splitLists =
+        items.stream().collect(Collectors.partitioningBy(
+          i -> FieldValues.CLASS_RESIDENTIAL.equals(i.attrs().get(Fields.CLASS)))
```

Comment:
No action necessary here, but in a separate review, we might want to make `Feature` implement `WithTags` over in planetiler so you can do `i.hasTag(Fields.CLASS, FieldValues.CLASS_RESIDENTIAL)`